cmd/util/ledger/reporters/account_reporter.go

```diff
@@ -92,7 +94,6 @@ func (r *AccountReporter) Report(payload []ledger.Payload) error {
 			panic(err)
 		}
 	}
-	close(addressIndexes)
```

Comment:
I think L86 `go adp.reportAccountData(addressIndexes)` was still sending data to the channel, but we are closing the channel too early. 

It should only be closed after the `Wait()`.

I used `defer` to ensure it's closed in the end

Reply:
reportAccountData  gets addressIndexes as a receive only channel so I think that channel is not the problem.

This close must be before `wg.Wait()` because this close causes all the `go adp.reportAccountData` to finish processing and exit (calling `wg.Done()` as they do). with this change `wg.Wait()` will never finish.

I suspect the problem is coming from the `ReportWriter` channel closing too early. However going through the code  I am still not sure how, since the report writer channels are the last to be closed: 
```
defer rwa.Close()
defer rwc.Close()
defer rwm.Close()
```

Reply:
The Waitgroup management is a bit confusing to me, because we are waiting for the worker to stop working, but it might run into concurrency issue where all worker finished their work but immediately pick up new work, however the wait group will consider as finished. And close all the channels including the writers channel, however, there might be some process still running `reportAccountData` in a goroutine, and when it writes to the writer channel, it will crash.

A more reliable way is to add all indexes to the wait group, and wait until the workers have finished the same number of jobs as the total number of address.

Reply:
That makes sense and looks much nicer!